File: Rest1/youquizcreation/src/main/java/ch/morisettid/youquizcreation/model/Quiz.java
package ch.morisettid.youquizcreation.model;

import java.util.List;

import org.json.JSONArray;
import org.json.JSONObject;

import org.apache.tomcat.util.json.JSONParser;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(name = "T_Quiz")
public class Quiz {

    // Attributs
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "PK_Quiz")
    private Integer pkQuiz;

    @Column(name = "Nom", length = 50)
    private String nom;

    @Column(name = "Description", length = 250)
    private String description;

    @OneToMany(mappedBy = "quiz")
    private List<Question> questions;

    // Getters et Setters
    public Integer getPkQuiz() {
        return pkQuiz;
    }

    public void setPkQuiz(Integer pkQuiz) {
        this.pkQuiz = pkQuiz;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return this.description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public List<Question> getQuestions() {
        return this.questions;
    }

    public void setQuestions(List<Question> questions) {
        this.questions = questions;
    }
}

File: Client1/js/beans/Quiz.js
class Quiz {
    constructor(pkQuiz, nom, description, username, questions) {
        this.pkQuiz = pkQuiz || null;
        this.nom = nom || '';
        this.description = description || '';
        this.username = username || '';
        this.likes = 0;
        this.questions = questions || [];
    }

    static fromJSON(quizJSON) {
        let questions = [];
        if (quizJSON.questions.length > 0) {
            quizJSON.questions.forEach(questionJSON => {
                questions.push(Question.fromJSON(questionJSON));
            });
        }
        return new Quiz(quizJSON.pkQuiz, quizJSON.nom, quizJSON.description, quizJSON.username, questions);
    }

    // Getters and setters...
    getPkQuiz() {
        return this.pkQuiz;
    }

    setPkQuiz(pkQuiz) {
        this.pkQuiz = pkQuiz;
    }

    getNom() {
        return this.nom;
    }

    setNom(nom) {
        this.nom = nom;
    }

    getDescription() {
        return this.description;
    }

    setDescription(description) {
        this.description = description;
    }

    getUsername() {
        return this.username;
    }

    setUsername(username) {
        this.username = username;
    }

    getLikes() {
        return this.likes;
    }

    setLikes(likes) {
        this.likes = likes;
    }

    getQuestions() {
        return this.questions;
    }

    setQuestions(questions) {
        this.questions = questions;
    }
}
